presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
         org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, handle.getSchemaTableName());
         return TableStatisticsMaker.getTableStatistics(typeManager, session, constraint, handle, icebergTable);
     }
+
+    @Override
+    public void createMaterializedView(ConnectorSession session, SchemaTableName viewName, ConnectorMaterializedViewDefinition definition, boolean replace, boolean ignoreExisting)
+    {
+        Map<String, String> properties;
+
+        Map buildProps = new HashMap();
+        buildProps.put(PRESTO_QUERY_ID_NAME, session.getQueryId());
+
+        // Generate a storage table name
+        String storageTableName = "st" + UUID.randomUUID().toString().replace("-", "");
+        buildProps.put(STORAGE_TABLE, storageTableName);
+        buildProps.put(PRESTO_VIEW_FLAG, "true");
+        if (replace && definition.getComment().isPresent()) {
+            buildProps.put(TABLE_COMMENT, definition.getComment().get());
+        }
+        else {
+            // Add a default table comment when the view is being created
+            buildProps.put(TABLE_COMMENT, "Presto Materialized View");
+        }
+        properties = ImmutableMap.<String, String>builder()
+            .putAll(buildProps)
+            .build();
+
+        io.prestosql.plugin.hive.metastore.Column dummyColumn = new io.prestosql.plugin.hive.metastore.Column("dummy", HIVE_STRING, Optional.empty());
+
+        String schemaName = viewName.getSchemaName();
+        String tableName = viewName.getTableName();
+        Table.Builder tableBuilder = Table.builder()
+                .setDatabaseName(schemaName)
+                .setTableName(tableName)
+                .setOwner(session.getUser())
+                .setTableType(org.apache.hadoop.hive.metastore.TableType.VIRTUAL_VIEW.name())
+                .setDataColumns(ImmutableList.of(dummyColumn))
+                .setPartitionColumns(ImmutableList.of())
+                .setParameters(properties);
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(VIEW_STORAGE_FORMAT)
+                .setLocation("");
+
+        HiveIdentity identity = new HiveIdentity(session);
+        Optional<Table> existing = metastore.getTable(identity, viewName.getSchemaName(), viewName.getTableName());
+
+        // It's a create command where the materialized view already exists and 'if not exists' clause is specified
```

Comment:
This seems to be missing the "not" word in "clause is specified"
```java
// create command where materialized view already exists and IF NOT EXISTS clause is not specified

Reply:
yes, correct, good catch